lint-rules-rxjava2-lint/src/test/java/com/vanniktech/lintrules/rxjava2/RxJava2MethodMissingCheckReturnValueDetectorSytemPropTest.kt

```diff
@@ -0,0 +1,118 @@
+    System.setProperty(IGNORE_MODIFIERS_PROP, "private")
+
+    lint()
+      .files(rxJava2(), java(
+          """
+          package foo;
+
+          import io.reactivex.Observable;
+
+          class Example {
+            private Observable<Object> foo() {
+              return null;
+            }
+          }"""
+        ).indented()
+      )
+      .issues(ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE)
+      .run()
+      .expectClean()
+  }
+
+  @Test fun multipleIgnoredModifiers() {
+    System.setProperty(IGNORE_MODIFIERS_PROP, "private,protected")
+
+    lint()
+      .files(rxJava2(), java(
+          """
+          package foo;
+
+          import io.reactivex.Observable;
+
+          class Example {
+            private Observable<Object> foo() {
+              return null;
+            }
+
+            protected Observable<Object> bar() {
+              return null;
+            }
+          }"""
+        ).indented()
+      )
+      .issues(ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE)
+      .run()
+      .expectClean()
+  }
+
+  @Test fun protectedMethodReturningObservableMissingCheckReturnValue() {
+    System.setProperty(IGNORE_MODIFIERS_PROP, "private")
+
+    lint()
+      .files(rxJava2(), java(
+        """
+          package foo;
+
+          import io.reactivex.Observable;
+
+          class Example {
+            protected Observable<Object> foo() {
+              return null;
+            }
+          }"""
+      ).indented()
+      )
+      .issues(ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE)
+      .run().expect("""
+          |src/foo/Example.java:6: Warning: Method should have @CheckReturnValue annotation. [RxJava2MethodMissingCheckReturnValue]
+          |  protected Observable<Object> foo() {
+          |                               ~~~
+          |0 errors, 1 warnings""".trimMargin())
+      .expectFixDiffs("""
+          |Fix for src/foo/Example.java line 6: Add @CheckReturnValue:
+          |@@ -6 +6
+          |-   protected Observable<Object> foo() {
+          |+   io.reactivex.annotations.CheckReturnValue protected Observable<Object> foo() {
+          |""".trimMargin())
+  }
+
+  @Test fun kotlinPrivateMethodReturningTestSubscriberMissingCheckReturnValue() {
+    System.setProperty(IGNORE_MODIFIERS_PROP, "private")
+
+    lint()
+      .files(rxJava2(), kt(
+          """
+            package foo
+
+            import io.reactivex.Observable
+
+            class Example {
+              private fun foo(): Observable<Object>? {
+                return null
+              }
+            }"""
+        ).indented()
+      )
+      .issues(ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE)
+      .run()
+      .expectClean()
+  }
+
+  @After fun tearDown() {
+    System.clearProperty(IGNORE_MODIFIERS_PROP)
+  }
+
+  companion object {
+    private const val IGNORE_MODIFIERS_PROP = "com.vanniktech.lintrules.rxjava2.RxJava2MethodMissingCheckReturnValueDetector.ignoreMethodAccessModifiers"
```

Comment:
Just make the const in the detector `internal` and reuse it

Reply:
Will do. Let me explain my intention here. 

In the future if the detector one is accidentally changed, it would be a breaking change and tests won't catch that since they will be reusing the modified one. 

Do you think that makes sense?

Reply:
Ah okay get it. Then leave it as it is now